runtime/interpreter/value.go

```diff
@@ -20311,6 +20311,17 @@ func (v *EphemeralReferenceValue) GetMember(
 ) Value {
 	self := v.MustReferencedValue(interpreter, locationRange)
 
+	switch name {
+	case sema.ReferenceTypeDereferenceFunctionName:
+		return NewHostFunctionValue(
+			interpreter,
+			sema.ReferenceDereferenceFunctionType(v.BorrowedType),
+			func(invocation Invocation) Value {
+				return v.Value
+			},
```

Comment:
The implementation of `dereference` looks straightforward. Is this correct or am I missing something?

Reply:
shouldn't this create new Value ?

Reply:
Yeah, this looks right. The copy should be done automatically by the interpreter. Maybe test this via, e.g. 

```cadence
let ints = [1, 2, 3]
let ref: &[Int] = &ints
let ints2 = *ref // not sure what the syntax will be
ints.append(4)
assert(ints == [1, 2, 3, 4])
assert(ints2 == [1, 2, 3])
ints2.append(5)
assert(ints == [1, 2, 3, 4])
assert(ints2 == [1, 2, 3, 5])
```

Reply:
I was thinking more like: 

```cadence
let x: &[Int] = &[1]
x.dereference().append(2)
``

Reply:
Yeah, this needs a `Transfer` to the stack, i.e. 
```go
.Transfer(
	interpreter,
	locationRange,
	atree.Address{},
	false,
	nil,
	nil,
)
```

Reply:
Great point @bluesign to test by operating on the returned value directly, my example above, `let ints2 = ref.dereference()` was adding a copy 👍